client/a8c-for-agencies/sections/sites/needs-setup-sites/site-configurations-modal/index.tsx

```diff
@@ -58,7 +60,42 @@ export default function SiteConfigurationsModal( { toggleModal }: SiteConfigurat
 						'Once the site is created, you can connect a custom domain to your site and make that your site address instead.'
 					) }
 				>
-					<FormTextInputWithAffixes suffix=".wpcomstaging.com" noWrap />
+					<div className="configure-your-site-modal-form__site-name-wrapper">
+						<FormTextInputWithAffixes
+							isError={ siteName.showValidationMessage }
+							value={ siteName.siteName }
+							onChange={ ( event: React.ChangeEvent< HTMLInputElement > ) =>
+								siteName.setSiteName( event.target.value )
+							}
+							disabled={ siteName.isRandomSiteLoading }
+							suffix=".wpcomstaging.com"
+							noWrap
+							spellcheck="false"
+							placeholder={ siteName.isRandomSiteLoading && '...' }
```

Comment:
It doesn't seem to be a nice experience to open the modal with the loading state on the first field. Maybe it could load the address suggestion before opening the modal or before displaying the fields. What do you think?

Reply:
Thank you Paulo!
I addressed in two steps:

1 - The random site name starts to load before user clicking on Create site.

2- If for some reason, like a slow internet connection, the user manages to click on create site before the loading is ready, we display the shimming animation on field:
![Screen Recording 2024-07-02 at 15 29 55](https://github.com/Automattic/wp-calypso/assets/33497086/95cd7fb1-a100-4208-b998-7affbf189058)

Reply:
Thanks for the fixes!